client/blocks/props-viewer/index.jsx

```diff
@@ -0,0 +1,192 @@
+	 * @param {string} rightName The prop name of the right side
+	 * @return {number} Which side wins
+	 */
+	sortProps = ( leftName, rightName ) => {
+		const component = this.state.component;
+
+		const left = component.props[ leftName ];
+		const right = component.props[ rightName ];
+		if ( left.required === right.required ) {
+			return ( leftName.toLowerCase() < rightName.toLowerCase() ? -1 : 1 );
+		}
+
+		if ( left.required ) {
+			return -1;
+		}
+
+		return 1;
+	};
+
+	/**
+	 * Renders a row in the table
+	 * @param {object} component The component
+	 * @param {string} propName The name of the prop to render
+	 * @return {ReactElement} The rendered row
+	 */
+	renderRow = ( component, propName ) => {
+		const prop = component.props[ propName ];
+		let type = 'unknown';
+		if ( prop.type ) {
+			switch ( prop.type.name ) {
+				default:
+					type = prop.type.name;
+					break;
+				case 'arrayOf':
+				case 'oneOf':
+				case 'oneOfType':
+				case 'objectOf':
+				case 'instanceOf':
+					type = `${ prop.type.name }( ${ prop.type.value.name || 'unknown' } )`;
+					break;
+			}
+		}
+
+		return (
+			<tr key={ propName }>
+				<td>{ prop.required ? <Gridicon icon="checkmark" /> : <Gridicon icon="cross-small" /> }</td>
+				<td>{ propName }</td>
+				<td>{ type }</td>
+				<td>{ prop.defaultValue ? prop.defaultValue.value : 'undefined' }</td>
+				<td>{ prop.description }</td>
+			</tr>
+		);
+	};
+
+	/**
+	 * Renders a table if it can
+	 * @param {object} component The component to render for
+	 * @return {ReactComponent|null} The table or nothing
+	 */
+	renderTable = ( component ) => {
+		if ( ! component ) {
+			return null; //todo: explain why this table is missing
+		}
+
+		return (
+			<Card compact={ true } className="props-viewer__card" tagName="div">
+				<p className="props-viewer__description" >{ component.description || 'Add jsdoc to the component see it here' }</p>
+				<div className="props-viewer__usage">
+					<div className="props-viewer__example" >
+						<span className="props-viewer__heading">Use It</span>
+						<p><code>
+							import { component.displayName } from '{ component.includePath }';
+						</code></p>
+					</div>
+					<div className="props-viewer__table">
+						<span className="props-viewer__heading">Props</span>
+						<table>
+							<thead>
+							<tr>
+								<td>Required</td>
+								<td>Name</td>
+								<td>Type</td>
+								<td>Default</td>
+								<td>Description</td>
+							</tr>
+							</thead>
+							<tbody>
+							{ component.props
+								? Object.keys( component.props )
+									.sort( this.sortProps )
+									.map( ( propName ) => this.renderRow( component, propName ) )
+								: null }
+							</tbody>
+						</table>
+					</div>
+				</div>
+			</Card>
+		);
+	};
+
+	render() {
+		const component = this.state.component;
+
+		return (
+			<div>
+				{ this.props.example }
+				{ this.renderTable( component ) }
+			</div>
+		);
+	}
+
+	static propTypes = {
```

Comment:
I normally see them outside the class at the bottom of the file. I put them at the bottom so that someone would easily find them out of habit, without searching.